src/odemis/acq/move.py

```diff
@@ -2262,3 +2720,157 @@ def _getCurrentAlignerPositionLabel(self) -> int:
+        # it's read-only, so we change it via _value
+        self.position._set_value(pos, force_write=True)
+
+        # update related mds
+        for comp in self.affects:
+            try:
+                if comp:
+                    md_pos = pos.get("x", 0), pos.get("y", 0)
+                    comp.updateMetadata({
+                        model.MD_POS: md_pos,
+                        model.MD_STAGE_POSITION_RAW: pos_dep}
+                        )
+            except Exception as e:
+                logging.error("Failed to update %s with new position: %s", a, e)
+
+        # update the SEM focus position when the stage is moved to compensate for linked behavior
+        # TODO: update the self.sem_eucentric_focus when the user manually focuses.
+        if self.pm.use_linked_sem_focus_compensation:
+            try:
+                ebeam_focus = model.getComponent(role="ebeam-focus")
+                # get the eucentric focus position from the metadata
+                self.sem_eucentric_focus = self._stage_bare.getMetadata()[model.MD_CALIB].get("SEM-Eucentric-Focus", 7.0e-3)
+                f = ebeam_focus.moveAbs({"z": self.sem_eucentric_focus})
+                f.result()
+            except Exception as e:
+                logging.error("Failed to update ebeam-focus with new position: %s", e)
+
+    def _updateSpeed(self, dep_speed):
+        """
+        update the speed VA based on the dependency's speed
+        """
+        # stage_speed = self.pm.to_sample_stage_from_stage_movement(dep_speed)
+        self.speed._set_value(dep_speed, force_write=True)
+
+    @isasync
+    def moveRel(self, shift: Dict[str, float], **kwargs) -> Future:
+        """
+        :param shift: The relative shift to be made
+        :param **kwargs: Mostly there to support "update" argument
+        """
+        # missing values are assumed to be zero
+        shift_stage = self.pm.from_sample_stage_to_stage_movement(shift)
+        logging.debug("converted relative move from %s to %s", shift, shift_stage)
+        return self._stage_bare.moveRel(shift_stage, **kwargs)
+
+    @isasync
+    def moveAbs(self, pos: Dict[str, float], **kwargs) -> Future:
+        """
+        :param pos: The absolute position to be moved to
+        :param **kwargs: Mostly there to support "update" argument
+        """
+
+        # if key is missing from pos, fill it with the current position
+        for key in self.axes.keys():
+            if key not in pos:
+                pos[key] = self.position.value[key]
+
+        # pos is a position, so absolute conversion
+        pos_stage = self.pm.from_sample_stage_to_stage_position(pos)
+        logging.debug("converted absolute move from %s to %s", pos, pos_stage)
+        return self._stage_bare.moveAbs(pos_stage, **kwargs)
+
+    @isasync
+    def moveRelChamberReferential(self, pos: Dict[str, float]) -> Future:
+        """Move the stage vertically in the chamber. This is non-blocking.
+        From OpenFIBSEM"""
```

Comment:
This function is quite difficult to grasp. It needs more info. In particular, when mentioning a *vertical* move, I think typically one would expect a "z" value. However, only "x" and "y" are used. This should be explained. Also, for relative moves, normally the argument is called "shift". Please use that name here too, because "pos" sounds too much like a "target position" / "absolute position".

Could you explain in the docstring how Y (in input) is related to a Z movement (in chamber referential)? Looking at the code, if rx = 0, then dy == 0 and dz == Y... which gives me the feeling that actually the two shifts input are x & z (in chamber referential). Is that correct? (if so, the key should be renamed "y" -> "z").

Reply:
I've updated it to take x,z input shift and added more info to the docstring. x,y refer to the fact that this movement is calculated form the image view. Givne where this is, it's clearer that it is x,z